tests/specs/bihu/verification.k

```diff
@@ -81,3 +83,49 @@ module VERIFICATION
     rule                 @canExtractThisYear(COLLECTED +Int BAL, NOW, START) +Int ((COLLECTED +Int BAL) -Int @remainingTokens(COLLECTED +Int BAL, NOW, START)) <Int pow256                => true  requires #accumulatedReleasedTokens(BAL, COLLECTED, START, NOW) >Int COLLECTED +Int 3 [simplification]
 
 endmodule
+
+module VERIFICATION-HASKELL [symbolic, kore]
+    imports EVM
+
+    rule X <Int Csstore(SCHED, _, _, _) => false
+      requires Gsload       < SCHED > <=Int X
+       andBool Gsstoreset   < SCHED > <=Int X
+       andBool Gsstorereset < SCHED > <=Int X
+      [simplification]
+
+    rule Csstore(SCHED, _, _, _) <=Int X => true
+      requires Gsload       < SCHED > <=Int X
+       andBool Gsstoreset   < SCHED > <=Int X
+       andBool Gsstorereset < SCHED > <=Int X
+      [simplification]
+
+    rule X -Int Y <Int Z  => X -Int Z <Int Y  [concrete(X, Z), symbolic(Y), simplification]
+    rule X <=Int Y -Int Z => Z <=Int Y -Int X [concrete(X, Y), symbolic(Z), simplification]
+
+endmodule
+
+module VERIFICATION-JAVA [symbolic, kast]
+    imports EVM
+
+ // Csstore rule
+
+    rule Csstore(SCHED, NEW, CURR, ORIG)
+      => #if CURR ==Int NEW orBool ORIG =/=Int CURR
+           #then Gsload < SCHED >
+           #else #if ORIG ==Int 0
+                   #then Gsstoreset < SCHED >
+                   #else Gsstorereset < SCHED >
+                 #fi
+         #fi
+      requires Ghasdirtysstore << SCHED >>
+      [simplification]
+
+    rule Csstore(SCHED, NEW, CURR, _ORIG)
+      => #if CURR ==Int 0 andBool NEW =/=Int 0
+           #then Gsstoreset < SCHED >
+           #else Gsstorereset < SCHED >
+         #fi
+      requires notBool Ghasdirtysstore << SCHED >>
+      [simplification]
```

Comment:
These rules are already present in evm.md. Is there a reason for duplicating them?

Reply:
Yes, it's because they are marked as `concrete` now, so they won't be simplified anymore in `evm.md`. But the Java backend needs these rules to apply, so we duplicate them here for the java backend.